airflow-core/src/airflow/models/serialized_dag.py

```diff
@@ -91,9 +91,11 @@ def resolve(self) -> dict[str, list[DagDependency]]:
                 if dep_type == "asset":
                     dag_deps.append(self.resolve_asset_dag_dep(dep_data))
                 elif dep_type == "asset-name-ref":
-                    dag_deps.extend(self.resolve_asset_name_ref_dag_dep(dep_data))
+                    if asset_dag_dep := self.resolve_asset_name_ref_dag_dep(dep_data):
+                        dag_deps.append(asset_dag_dep)
                 elif dep_type == "asset-uri-ref":
-                    dag_deps.extend(self.resolve_asset_uri_ref_dag_dep(dep_data))
+                    if asset_dag_dep := self.resolve_asset_uri_ref_dag_dep(dep_data):
+                        dag_deps.append(asset_dag_dep)
```

Comment:
I like the previous `extend` code better tbh. Maybe change the resolve function to yield the DagDependency instead if there’s one, instead of returning.

Reply:
Sounds good. let me update it